src/hooks/useOutputMaterial.ts

```diff
@@ -52,59 +87,74 @@ export function useOutputMaterial(sessionId: string, outputIdOrName: string, mat
 	 */
 	outputNode: ITreeNode | undefined
 } {
-
-	const materialPropertiesRef = useRef(materialProperties);
+	const materialPropertiesRef = useRef<IMaterialAbstractDataProperties | undefined>(undefined);
+	const materialTypeRef = useRef<MaterialType | undefined>(undefined);
 	
-	// callback which will be executed on output update, and once the node is available
-	const callback = useCallback( (node?: ITreeNode) => {
-		const materialProps = materialPropertiesRef.current;
-		if (!node)
-			return;
-
-		// get all geometry and materials below the node
-		const geometryData = getGeometryData(node);
-		const materials = getMaterials(geometryData);
+	// callback which will be executed on update of the output node
+	const callback = useCallback( (newNode?: ITreeNode, oldNode?: ITreeNode) => {
 	
-		if (materials.length > 0) {
-			// update existing materials
-			materials.forEach((m) => {
-				for ( const p in materialProps)
-					(<any>m)[p as keyof MaterialStandardData] =
-					materialProps[p as keyof IMaterialStandardDataProperties];
-				m.updateVersion();
+		// restore original materials
+		// TODO there seems to be a bug, oldNode.id should not be a new id
+		console.debug("newNode.id", newNode?.id, "oldNode?.id", oldNode?.id, originalMaterials);
```

Comment:
@MajorMeerkatThe3rd there seems to be some bug related to the output update callback: The "old" node does not have the same id it had when it was the new node 😁

Reply:
Related task: https://shapediver.atlassian.net/browse/SS-7366